### ISSUE TITLE
should not get typo results if a match is found

### ISSUE THREAD

**Issue by jhoguet** [typesense/typesense#304]
## Description

I am not expecting typo based results when setting typo_tokens_threshold to 1 and there is a match

## Steps to reproduce

```
  const schemaName = 'sports-bugs-v1'
    await typesense.collections().create({
        "name": schemaName,
        "fields": [
            { "name": "DESCR", "type": "string" },
            { "name": "SPORT_CODE", "type": "string" },
        ],
    });

    await typesense.collections(schemaName).documents().import([
        { DESCR: 'equestrian sports', SPORT_CODE: 'EQUES', id: '64' },
        { DESCR: 'pom squad', SPORT_CODE: 'POM S', id: '127' },
        { DESCR: 'dance squad', SPORT_CODE: 'DANC2', id: '57' }
    ]);

    const result = await typesense.collections(schemaName).documents().search({
        'query_by': 'DESCR',
        'q': 'eque',
        per_page: 10,
        limit_hits: 10,
        typo_tokens_threshold: 1
    })
```
## Expected Behavior

I expect results to contain `equestrian sports` and no results based on typos since typo_tokens_threshold is 1 and we would need less than 1 to include typo based results. 

## Actual Behavior

I get `equestrian sports` as expected, but I also get `dance squad` and `pom squad`

Note, that if I set `typo_tokens_threshold` to 0, then I only get `equestrian sports`

## Metadata

This is reproducing on 0.20 and 0.21.0.rc14

**Comment by kishorenc** [comment]
@jhoguet This is indeed an edge case that we must address. 

This is happening because, by default Typesense indexes the documents into 4 memory shards. Since there is only one result and that document exists in a shard, in the other 3 shards, typo correction happens. When the results from all the shards are then aggregated, the other hits are showing up. We need to do another filtering at the final aggregation level to address this.

As a work around, you can create a collection with a single memory shard using the `num_memory_shards ` parameter, like this:

```
curl -k "http://localhost:8108/collections" -X POST -H "Content-Type: application/json" \ 
      -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" -d '{
        "name": "titles", 
        "num_memory_shards": 1,
        "fields": [
          {"name": "title", "type": "string" },
          {"name": "points", "type": "int32" }
        ]
      }'
```

This should not have any performance impact unless you're indexing multiple millions of documents.

**Comment by kishorenc** [comment]
Fixed in `v0.22`.
